src/Accelerators/NNPA/Conversion/ZLowToLLVM/ZLowToLLVM.cpp

```diff
@@ -1281,11 +1253,8 @@ class ZLowPool2DLowering : public ConvertToLLVMPattern {
     Value input = operands[0];
     Value shape = operands[1];
     Value output = operands[2];
-    Type llvmElementTy = input.getType()
-                             .dyn_cast<LLVM::LLVMStructType>()
-                             .getBody()[0]
-                             .cast<LLVM::LLVMPointerType>()
-                             .getElementType();
+    Type llvmElementTy = typeConverter->convertType(
+        op->getOperand(0).getType().cast<MemRefType>().getElementType());
```

Comment:
I have to keep `op->getOperand` in lowering templates such unary, binary and pooling, since each template is used for multiple ops, so `getOperand(0)` is convenient. But I tried to operand names it in other places as many as possible.